### PR TITLE
Make shard cleanup query more efficient

### PR DESCRIPTION
**Pull request by AlekseiS**
Note, that you need an index on (end_time, transaction_id) for
"transactions" table.

### CODE REVIEW

**Comment by electrum** [comment]
Remove the description

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/H2ShardDao.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
     @SqlUpdate("DELETE FROM transactions\n" +
             "WHERE end_time < :maxEndTime\n" +
-            "  AND successful IS NOT NULL\n" +
+            "  AND successful IN (TRUE, FALSE)\n" +
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/H2ShardDao.java]: This is not needed for H2

**Comment by AlekseiS** [comment]
Addressed comments and added deletion in a loop

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java [hunk 2]
```diff
@@ -298,7 +298,12 @@ void deleteOldShards()
     @VisibleForTesting
     void deleteOldCompletedTransactions()
     {
-        dao.deleteOldCompletedTransactions(maxTimestamp(maxCompletedTransactionAge));
+        while (!Thread.currentThread().isInterrupted()) {
+            int deleted = dao.deleteOldCompletedTransactions(maxTimestamp(maxCompletedTransactionAge));
+            if (deleted < ShardDao.CLEANUP_TRANSACTIONS_BATCH_SIZE) {
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java]: Static import

Comment (AlekseiS) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java]: Done

**Pull request merged**